taxi/src/lib.rs

```diff
@@ -21,18 +19,20 @@ pub enum Dir {
 }
 
 impl Dir {
-	pub fn from_u32(int: &u32) -> Dir {
-		match *int {
-			0 => Dir::Up,
-			1 => Dir::Right,
-			2 => Dir::Down,
-			_ => Dir::Left,
+	/// Converts u32 to `Dir`
+	pub fn from_u32(int: u32) -> Result<Dir, &'static str> {
+		match int {
+			0 => Ok(Dir::Up),
+			1 => Ok(Dir::Right),
+			2 => Ok(Dir::Down),
+			3 => Ok(Dir::Left),
+			_ => Err("Cannot convert u32 to `Dir`"),
 		}
 	}
 }
 
 #[derive(PartialEq, Eq, Hash, Clone, Copy)]
-pub enum Object {
+enum Object {
```

Comment:
`Object` was public but not possible to use with `Game` interface. Just unnecessary distraction